Lib/test/test_winreg.py

```diff
@@ -340,7 +339,22 @@ def test_setvalueex_value_range(self):
                 SetValueEx(ck, "test_name", None, REG_DWORD, 0x80000000)
         finally:
             DeleteKey(HKEY_CURRENT_USER, test_key_name)
-
+    
+    def test_setvalueex_negative_one_check(self):
+        # Test for Issue #43984, check -1 was not set by SetValueEx.
+        # Py2Reg, which gets called by SetValueEx, wasn't checking return
+        # value by PyLong_AsUnsignedLong, thus setting -1 as value in the registry.
+        # The implementation now checks PyLong_AsUnsignedLong return value to assure
+        # the value set was not -1.
+        try:
+            with CreateKey(HKEY_CURRENT_USER, test_key_name) as ck:
+                with self.assertRaises(OverflowError):
+                    SetValueEx(ck, "test_name", None, REG_DWORD, -1)
+                with self.assertRaises(FileNotFoundError):
+                    QueryValueEx(ck, "test_name")
```

Comment:
1. You should assert that the problem you are addressing in the PR is fixed: did `SetValueEx` overwrite the current registry value or not? Currently, you do not assert that the queried value is unchanged.
2. There's no test case for `REG_QWORD`.
3. Nit: I'd shorten the docstring to a one-liner: _"bpo-43984: assert that registry values are not set to -1 in the case of overflow errors."_

Reply:
> 1. Queried value wouldn't exist because SetValueEx didn't create or set the key. The assert case here is QueryValueEx would raise a FileNotFoundError because it couldn't find the key.

Makes sense. Sorry 'bout the noise.

Reply:
> Queried value wouldn't exist because SetValueEx didn't create or set the key. The assert case here is QueryValueEx would raise a FileNotFoundError because it couldn't find the key.

`SetValueEx` and `QueryValueEx` are setting and querying a value, not a key. 

The analog to a filesystem is that a key is like a directory; a value is like a [named stream](https://docs.microsoft.com/en-us/windows/win32/fileio/file-streams) in a directory (e.g. "dirname:streamname:$DATA"); and the value type (e.g. `REG_SZ`) is like a stream type (e.g. $DATA). A directory can be mounted by another filesystem, or a directory can have a symlink stream (i.e. a symlink $REPARSE_POINT stream) that targets another directory. Similarly a key can be mounted by a hive (e.g. "\\Registry\\Machine\\System" is mounted by the hive "%SystemRoot%\\System32\\config\\SYSTEM"), or a key can contain a symlink (`REG_LINK` value type) to another key (e.g. "\\Registry\\User\\\<sid\>\\Software\\Classes" is a symlink to "\\Registry\\User\\\<sid\>_Classes"). Values in a key are not individually secured by a security descriptor, just like named streams in a directory are not individually secured. A significant difference is that values are not accessed as objects that can be the target of an open, unlike named streams in a directory. Thus a value cannot be referenced by a handle, and a value cannot be the target of a symlink.

Reply:
Ooops. I always mess up between SetValueEx, QueryValueEx  and SetValue, QueryValue. Anyway the value will never be created and set by SetValueEx. That's what I meant to say.

Reply:
Thanks for the explanation, Eryk!

Reply:
I'm adding the test for REG_QWORD now.